annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/field/FieldDeclarationStore.java

```diff
@@ -136,4 +140,22 @@ public OnField getLocationOnField(String clazz, String field) {
     }
     return new OnField(candidate.pathToSourceFile, candidate.clazz, fieldNames);
   }
+
+  /**
+   * Creates a {@link edu.ucr.cs.riple.injector.location.OnClass} instance targeting the passed
+   * classes flat name.
+   *
+   * @param clazz Enclosing class of the field.
+   * @return {@link edu.ucr.cs.riple.injector.location.OnClass} instance targeting the passed
+   *     classes flat name.
+   */
+  public OnClass getLocationOnClass(String clazz) {
+    FieldDeclarationInfo candidate =
+        findNodeWithHashHint(node -> node.clazz.equals(clazz), FieldDeclarationInfo.hash(clazz));
+    if (candidate == null) {
+      // field is on byte code.
```

Comment:
So, this comment (and the name of the class), reference a `field`, but as far as I can tell this is looking for a class declaration and there is no field involved...

Reply:
Yes, this class has the required information which its main usage is to store information regarding fields. I have a plan for refactoring these data structures that store information regarding the structure of a code and combine them into one data structure. But the plan is to do that after the release of `1.3.6`. Please let me know if you would like me to change it. Thank you.